### PR TITLE
tests: add tests with github actions (kind) & Makefile: remove 'test-with-kind'

### PR DESCRIPTION
**Pull request by hj-johannes-lee**
<Commit 1> Makefile: add target 'install-tools'

Target 'install-tools' installs:
- controller-gen
- golangci-lint
- kind

Closes: #623

<Commit 2> Makefile: update target 'test-with-kind'

- Update target to use go for installing KinD
- Use Docker to build images
- Add e2e test for SGX admission webhook
- Add deployment of oeprator
- Add a test for operator deployment

<Commit 3> tests: add tests with github actions (test-with-kind)
Closes: #613

### CODE REVIEW

**Comment by bart0sh** [comment]
@hj-johannes-lee can you explain the reason for removing Makefile target? What should I do if I want to run this locally?

**Comment by hj-johannes-lee** [comment]
@bart0sh Well, I first saw that it is somewhat outdated. As you can see, the Makefile has only test for fpga webhook. So, when I asked Mikko about this matter, he said we can just drop the target.

If you think it would be better to just make it up-to-date with sgx webhook (would there be something else than this?), let me try.

@mythi What do you think?

**Comment by mythi** [comment]
> Mikko about this matter, he said we can just drop the target.

right, it was with the assumption that we get `kind` install from of the actions. `test-with-kind` is outdated and hard to use. As we now install `kind` cluster manually we can of course update `test-with-kind` using the steps in this PR and make the new action just call `make test-with-kind`.

**Review by bart0sh — APPROVED** [review]
/lgtm

File: Makefile [hunk 1]
```diff
@@ -67,17 +64,20 @@ else
 endif
 
 test-with-kind: fixture
```
Comment (mythi) [inline Makefile:66]:
```suggestion
test-with-kind: fixture intel-sgx-admissionwebhook intel-fpga-admissionwebhook intel-deviceplugin-operator
```

would this work?

File: Makefile [hunk 2]
```diff
@@ -67,17 +64,20 @@ else
...
-	@$(GO) test -v ./test/e2e -args -kubeconfig $(e2e_tmp_dir)/kubeconfig -kubectl-path $(KUBECTL) -ginkgo.focus "FPGA Admission" || rc=1; \
-	$(KIND) delete cluster --name "intel-device-plugins"; \
-	rm -rf $(e2e_tmp_dir); \
-	exit $$rc
+	# Build Images
+	make intel-sgx-admissionwebhook
+	make intel-fpga-admissionwebhook
+	make intel-deviceplugin-operator
+	# Build a Cluster with KinD & Install Cert-Manager 
+	go get sigs.k8s.io/kind@${KIND_VERSION}
```
Comment (mythi) [inline Makefile:72]: I'm thinking we could we have some `install-tools`  target where we put all these. See also #623

Comment (hj-johannes-lee) [reply Makefile:72]: Got it, I assigned #623 to myself! :)

**Comment by hj-johannes-lee** [comment]
Closes: #623

**Review by mythi — DISMISSED** [review]

**Comment by bart0sh** [comment]
@hj-johannes-lee can you fix failing CI/kind-test ?

**Review by mythi — APPROVED** [review]

**Review by bart0sh — APPROVED** [review]
/lgtm

**Pull request merged**